cmd, common, node, rpc: move IPC into the node itself

karalabe opened the pull request:
**Edit: the HTTP and WebSocket API endpoints are currently implemented as singletons. Because of this, it will take a much larger effort to move them into the node. Hence this PR will only do the IPC part, and I'll leave the HTTP to a follow up PR with a more constrained scope.**

This PR supersedes https://github.com/ethereum/go-ethereum/pull/2031. The goal is to have the RPC managed fully by the node, configurable by some config parameters. This is particularly needed because currently the RPC API internals are spilled all over the node-client code (e.g. [etherapis](https://github.com/gophergala2016/etherapis/blob/master/etherapis/geth/geth.go#L95)).

 - [x] Move the IPC RPC endpoint fully into the `node`
 - ~~[ ] Move the HTTP RPC endpoint fully into the `node`~~
 - ~~[ ] Move the WS RPC endpoint fully into the `node`~~

robotally commented:
| Vote | Count | Reviewers |
| :---: | :---: | :---: |
| :+1: | 0 |  |
| :-1: | 0 |  |

_Updated: Tue Feb 2 17:09:13 UTC 2016_

obscuren commented on node/node.go:
```diff
@@ -172,6 +187,63 @@ func (n *Node) Start() error {
+func (n *Node) startRPC(services map[reflect.Type]Service) error {
+	// Gather and register all the APIs exposed by the services
+	ipcHandler := rpc.NewServer()
+	httpHandler := rpc.NewServer()
```
Shouldn't this only be done when we actually supply the `--rpc` flag?

karalabe replied:
Ah, indeed that was a leftover from when I still wanted to do the HTTP in this PR. Will remove the http version from this one.

obscuren commented:
Apart from the one http comment LGTM :+1:

karalabe commented:
@obscuren PTAL. I've deleted the `httpHandler` notions. Also I've moved the `ipcHandler` into a member var of the `node`. This allows gracefully terminating the live IPC connections instead of just destroying the server from underneath them.
